components/dialogs/AuthDialog.vue

```diff
@@ -872,6 +881,36 @@ export default {
         };
       }
 
+      if (this.signInPayload.email) {
+        const RANDOM_DIGIT_LENGTH = 5;
+        const MAX_SUGGEST_TRY = 5;
+        let [suggestedName] = this.signInPayload.email.split('@');
+        suggestedName = suggestedName.trim().toLowerCase().replace(/[^a-z0-9-_]/g, '-');
+        suggestedName = suggestedName.substring(0, MAX_USER_ID_LENGTH - RANDOM_DIGIT_LENGTH);
+        let isIDAvailable = false;
+        let tries = 0;
+        let tryName = suggestedName;
+        if (suggestedName.length < MIN_USER_ID_LENGTH) {
+          tryName = `${suggestedName}${String(Math.ceil(Math.random() * 10000)).padStart(RANDOM_DIGIT_LENGTH, '0')}`;
+        }
+        while (!isIDAvailable && tries < MAX_SUGGEST_TRY) {
+          try {
+            await apiGetUserMinById(tryName); // eslint-disable-line no-await-in-loop
+          } catch (err) {
+            if (err.response) {
+              if (err.response.status === 404) {
+                isIDAvailable = true;
+              }
+            }
+          }
+          tryName = `${suggestedName}${String(Math.ceil(Math.random() * 10000)).padStart(RANDOM_DIGIT_LENGTH, '0')}`;
```

Comment:
`tryName` is always changed even if `isIDAvailable` is true

Reply:
Also, should the harcoded `10000` depends on the value of `RANDOM_DIGIT_LENGTH`, say `Math.pow(10, RANDOM_DIGIT_LENGTH)`?

Reply:
Updated to use`break` and `Math.floor` + `**`